### ISSUE TITLE
Cannot read property 'create' of undefined when creating via POST

### ISSUE THREAD

**Issue by gazzer82** [tjwebb/sails-permissions#28]
Hi,

I'm trying to get this setup on a completely fresh install, but whenever i try to create any new objects via a POST request i get the following errors logged in the console:

```
error: Sending 500 ("Server Error") response: 
 TypeError: Cannot read property 'create' of undefined
    at module.exports (/Users/garethjeanne/Documents/Programming/Heroku/htl/node_modules/sails-permissions/api/policies/AuditPolicy.js:9:26)
    at routeTargetFnWrapper (/usr/local/lib/node_modules/sails/lib/router/bind.js:179:5)
    at callbacks (/usr/local/lib/node_modules/sails/node_modules/express/lib/router/index.js:164:37)
    at param (/usr/local/lib/node_modules/sails/node_modules/express/lib/router/index.js:138:11)
    at pass (/usr/local/lib/node_modules/sails/node_modules/express/lib/router/index.js:145:5)
    at nextRoute (/usr/local/lib/node_modules/sails/node_modules/express/lib/router/index.js:100:7)
    at callbacks (/usr/local/lib/node_modules/sails/node_modules/express/lib/router/index.js:167:11)
    at /usr/local/lib/node_modules/sails/lib/router/bind.js:187:7
    at /Users/garethjeanne/Documents/Programming/Heroku/htl/node_modules/sails-permissions/api/policies/ModelPolicy.js:33:7
    at tryCatcher (/usr/local/lib/node_modules/sails/node_modules/waterline/node_modules/bluebird/js/main/util.js:24:31)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/sails/node_modules/waterline/node_modules/bluebird/js/main/promise.js:454:31)
    at Promise._settlePromiseAt (/usr/local/lib/node_modules/sails/node_modules/waterline/node_modules/bluebird/js/main/promise.js:530:18)
    at Promise._settlePromises (/usr/local/lib/node_modules/sails/node_modules/waterline/node_modules/bluebird/js/main/promise.js:646:14)
    at Async._drainQueue (/usr/local/lib/node_modules/sails/node_modules/waterline/node_modules/bluebird/js/main/async.js:79:16)
    at Async._drainQueues (/usr/local/lib/node_modules/sails/node_modules/waterline/node_modules/bluebird/js/main/async.js:89:10)
    at Async.drainQueues (/usr/local/lib/node_modules/sails/node_modules/waterline/node_modules/bluebird/js/main/async.js:14:14)
    at process._tickDomainCallback (node.js:381:11) [TypeError: Cannot read property 'create' of undefined]
```

Any idea what's going on, everything is default and i've followed the instructions as best i can.

If i do a get i can't see any objects in the permission, role or user endpoints at all.

Thanks

Gareth

**Comment by tjwebb** [comment]
I'll check it out

**Comment by gazzer82** [comment]
Great, thanks. 

I've just done a clean install, before installing sails-permissions all works as expected, afterwards error as above. It also seems to cause the models to be recreated every time I start sails, which doesn't seem correct?

**Comment by tjwebb** [comment]
yes there's another issue about the models being re-created. I think I'm using `create` instead of `findOrCreate` somewhere.

**Comment by cratervale** [comment]
+1

**Comment by RGBKnights** [comment]
I had to remove the AuditPolicy until this issue has been resolved.

**Comment by tjwebb** [comment]
This should be resolved in 1.0.12. Basically RequestLog and SecurityLog were missing from the generator. Re-run `sails generate permissions-api` and it will create the models.
